tests/e2e/utils.py

```diff
@@ -49,6 +49,18 @@ def is_min_2_3_1():
     return unittest.skipUnless(is_min_2_3_1(), "test torch 2.3.1")(test_case)
 
 
+def require_torch_2_5_1(test_case):
+    """
+    Decorator marking a test that requires torch >= 2.3.1
+    """
+
+    def is_min_2_5_1():
+        torch_version = version("torch")
+        return torch_version >= "2.5.1"
```

Comment:
using string checks for version isn't really safe. You should use something like `from packaging import version` with `version.parse(...)`

Reply:
see https://github.com/huggingface/transformers/blob/main/src/transformers/pytorch_utils.py#L20-L39

Reply:
fixed